deepchecks/vision/checks/performance/simple_model_comparison.py

```diff
@@ -109,39 +109,13 @@ def __init__(self,
                                            '["best", "worst"], metric_to_show_by must be specified.')
 
         self.metric_to_show_by = metric_to_show_by
-        self._state = {}
+        self._test_metrics = None
 
     def initialize_run(self, context: Context):
         """Initialize the metrics for the check, and validate task type is relevant."""
         context.assert_task_type(TaskType.CLASSIFICATION)
 
-        self._state[DatasetKind.TEST.value] = get_scorers_list(context.train, self.alternative_metrics)
-        self._state['Simple Model'] = get_scorers_list(context.train, self.alternative_metrics)
-
-        if context.train.task_type == TaskType.CLASSIFICATION:
-            class_prior = np.zeros(context.train.num_classes)
-            n_samples = 0
-            for label, total in context.train.n_of_samples_per_class.items():
-                class_prior[label] = total
-                n_samples += total
-            class_prior /= n_samples
-
-            if self.strategy == 'most_frequent':
-                dummy_prediction = np.zeros(context.train.num_classes)
-                dummy_prediction[np.argmax(class_prior)] = 1
-                self._state['dummy_prediction_generator'] = lambda: torch.from_numpy(dummy_prediction)
-            elif self.strategy == 'prior':
-                self._state['dummy_prediction_generator'] = lambda: torch.from_numpy(class_prior)
-            elif self.strategy == 'stratified':
-                self._state['dummy_prediction_generator'] = \
-                    lambda: torch.from_numpy(np.random.multinomial(1, class_prior))
-            elif self.strategy == 'uniform':
-                self._state['dummy_prediction_generator'] = \
-                    lambda: torch.from_numpy(np.ones(context.train.num_classes) / context.train.num_classes)
-            else:
-                raise DeepchecksValueError(
-                    f'Unknown strategy type: {self.strategy}, expected one of {_allowed_strategies}.'
-                )
+        self._test_metrics = get_scorers_list(context.train, self.alternative_metrics)
```

Comment:
Can you just add a simple if to check whether the strategy is valid?

Reply:
This `if` exists in the init